### ISSUE TITLE
The content-type is not JSON compatible on revokeAll invocation

### ISSUE THREAD

**Issue by jfbelisle** [lelylan/simple-oauth2#350]
#### Context

* *node version*: 14.15.0
* *module version with issue*: 4.1.0
* *last module version without issue*: ?
* *environment* (e.g. node, browser, native): nodeJS
* *any other relevant information*: Using MS Graph endpoints

#### What are you trying to achieve or the steps to reproduce?

I try to invoke the revokeAll function using an accessToken instance with the following code on the MS Graph endpoints

```js
const revokeAllToken = async (req) => {
  const cachedSession = req.session.microsoft;
  if (cachedSession) {
    const cachedAccessTokenInstance = await oauth2Client.createToken(cachedSession);
    try {
      await cachedAccessTokenInstance.revokeAll();
    } catch (error) {
      console.log('Error revoking token: ', error.message);
    }
  }
};
```

#### What was the result you got?
I receive the error "The content-type is not JSON compatible"

I used de DEBUG flag to try to find what's going on, here is the request:

```
simple-oauth2:request-options Using header authentication. Authorization header set to XXXXXXXXXX
simple-oauth2:request-options Using form request format
simple-oauth2:client Creating request to: (POST) mytenant.onmicrosoft.com/oauth2/v2.0/revoke
simple-oauth2:client Using request options: {
"headers":{
"Authorization":"Basic XXXXXXXXXX",
"Content-Type":"application/x-www-form-urlencoded"},
"payload":"token=XXXXXXXX&token_type_hint=access_token"}
 ```

#### What result did you expect?
It looks like there is a mismatch between the content-type sent by simple-oauth2 and the requested content-type by MSGraph. The other POST request for tokens work properly with the same "Content-Type":"application/x-www-form-urlencoded" header. 

So why is the revokeAll POST does not?

**Comment by jonathansamines** [comment]
@jfbelisle That's likely just a strict requirement on the MSGraphs side. Unfortunately though, we don't currently support overriding the http headers without changing it for the whole client. You might want to give the [bodyFormat option](https://github.com/lelylan/simple-oauth2/blob/master/API.md#options) (json) a try and see if that option works consistently for all MSGraphs auth endpoints. Otherwise, I am open to add support for it if anyone has the time to implement it.

**Comment by vazgabriel** [comment]
I have the same issue using Hydra OAuth

**Comment by jonathansamines** [comment]
Solved by #354.
